src/can.rs

```diff
@@ -0,0 +1,125 @@
+//! # Controller Area Network (CAN) Interface
+//!
+//!
+//! TX: Alternate Push-Pull Output
+//! RX: Input Floating Input
+//!
+//! ### CAN1
+//!
+//! | Function | Pins  |
+//! |----------|-------|
+//! | TX       | PB9   |
+//! | RX       | PB8   |
+//!
+//! ### CAN2
+//!
+//! | Function | Pins  |
+//! |----------|-------|
+//! | TX       | PB13  |
+//! | RX       | PB12  |
+
+use crate::bb;
+use crate::gpio::{
+    gpioa::{PA11, PA12},
+    gpiob::{PB12, PB13, PB5, PB6, PB8, PB9},
+    Alternate, AF9,
+};
+use crate::pac::{CAN1, CAN2};
+use crate::rcc::APB1;
+
+mod sealed {
+    pub trait Sealed {}
+
+    /// Bus associated to peripheral
+    pub trait RccBus {
+        /// Bus type;
+        type Bus;
+    }
+}
+
+pub trait Pins: sealed::Sealed {
+    type Instance;
+}
+
+/*
+    order: tx, rx similar to serial
+*/
+
+macro_rules! pins {
+    ($($PER:ident => ($tx:ident, $rx:ident),)+) => {
+        $(
+            impl sealed::Sealed for ($tx<Alternate<AF9>>, $rx<Alternate<AF9>>) {}
+            impl Pins for ($tx<Alternate<AF9>>, $rx<Alternate<AF9>>) {
+                type Instance = $PER;
+            }
+        )+
+    }
+}
+
+pins! {
```

Comment:
Quick look at the stm32f407 datasheet shows that the can signals can be mapped to even more pins (for larger packages).